### ISSUE TITLE
Calendar block: previous month should show calendar too

### ISSUE THREAD

**Issue by daria2303** [Automattic/wp-calypso#87383]
### What

The Calendar block shows all days of the month. At the bottom, there is a link to the previous month. When we click on it we get a list of the posts from that month but more intuitive and functional would be to display previous month calendar days. 

### Why

It is more intuitive and functional. 

### How

E.g. 
https://github.com/Automattic/wp-calypso/assets/11535091/723c79cb-86a2-4cf0-a6cb-773cabbd7c54

When I click on June I would expect to see the June calendar, not all posts from June. I would also like to be able to click on the previous month from there and so on.

**Comment by maciejpilarski** [comment]
📌 REPRODUCTION RESULTS
– Tested on Atomic – Replicated / Could Not Replicate / Uncertain

📌 FINDINGS/SCREENSHOTS/VIDEO
I was able to reproduce the issue on my test site:
![Markup on 2024-02-16 at 17:09:40](https://github.com/Automattic/wp-calypso/assets/9025643/d1be43aa-b076-45a7-8198-d0f17cc28b06)

📌 ACTIONS
– Triaged

**Comment by xavier-lc** [comment]
I agree that it's more intuitive that the calendar paginates on itself and not on monthly posts, but changing it's behaviour now would be confusing for users, I think. What's your take on this, @Automattic/dotcom-design ?

**Comment by fditrapani** [comment]
Thanks for the ping! 

As stated in the issue, I would also expect to see the calendar for that month. Not the posts for the months. 

Is the Calendar block unique to dotcom or is it a core block?

**Comment by xavier-lc** [comment]
> Is the Calendar block unique to dotcom or is it a core block?

It's a core block @fditrapani

**Comment by jasmussen** [comment]
Indeed, this is an upstream core issue. But I also believe it's one that cannot be fixed in the block. Here's what happens:

* The Calendar block just shows the calendar where the block is inserted.
* The Previous link takes you to an archive permalink for the previous month (i.e. WordPress would offer slugs like /year/month/day/post-slug, the previous link would take you to /year/month/ and show you all posts from that month).
* By clicking the "Previous" button, it takes you to the _Archive_ template. This is a separate template from wherever the Calendar block was inserted. I.e. it may not be inserted there at all. Or if it is, then indeed the calendar _will_ be shown when you click the previous month.

So this is not a dotcom issue, and it's unclear whether it can be fixed in core. Perhaps it can be suggested a best practice for theme developers to include a calendar block in Archive templates.
